Changes in VB Blocks & Bug Fixes & Translation

Sal4iDev opened the pull request:
- Added `Deep` in `[Config] Config Keys`
- Changed `ExpressionParameter` with label `GUI` in `[Player] Open GUI` to `InputParameter` with label `GUI ID`

Sal4iDev committed:
Some Changes

- Added `Deep` in `[Config] Config Keys`
- Changed ExpressionParameter `GUI` in `[Player] Open GUI` to `InputParameter` with label `GUI ID`

OfficialDonut commented:
- Put parameters on the same line in `new Block(...)` so the style is consistent across all classes.
- The GUI ID field is more flexible as an ExpressionParameter. Someone may want to use an arbitrary string expression as the ID e.g. a command argument.

Sal4iDev commented:
however, in the Create GUI component, in the GUI ID field, InputParameter is set. Then you should add an ExpressionParameter there so that you can also use different values ​​there?

Sal4iDev commented:
> however, in the Create GUI component, in the GUI ID field, InputParameter is set. Then you should add an ExpressionParameter there so that you can also use different values ​​there?

although I still don't see the point

OfficialDonut commented:
It makes sense for the ID to be hardcoded when defining the GUI, but forcing the ID to be hardcoded when opening is unnecessarily restrictive.

A contrived example:
```
command
  openGui(arg0)
```
vs
```
command
  if arg0 == "gui1"
    openGui(gui1)
  else if arg0 == "gui2"
    openGui(gui2)
  else if arg0 == "gui3"
    openGui(gui3)
```

Sal4iDev commented:
does it throw an error if the id is not valid? then you need to do this:

```
command
  attempt execution 
    openGui(arg0)
```

okay.

OfficialDonut commented:
Yes it would throw an exception if the id was invalid. Attempt execution is an option but letting an exception get thrown isn't best practice, instead you could do input validation by having a list of GUI IDs and checking if the list contains it.

Sal4iDev commented:
why is it better to check through the list?

OfficialDonut commented:
Explicitly checking things better conveys your intent and you dont have to rely on the implementation of the open gui code e.g. what if the code changes so it silently fails instead of throwing an exception?

Sal4iDev committed:
Update

- Added `ExprPluginAuthors`
- Added `ExprPluginName`
- Other less interesting changes

Sal4iDev committed:
Update StatStopLoop.java

- Added `StatAdvancedNumberLoop.class` as Valid for `Stop Loop` statement

Sal4iDev commented:
![image](https://user-images.githubusercontent.com/96550478/150493730-63a7cdba-d2e6-4162-adf0-d4695509a38f.png)
Done. Also now `Stop Loop` will work in `Advanced Number Loop`

Sal4iDev committed:
Update StatContinueLoop.java

Sal4iDev commented:
And `Continue Loop` will work too

OfficialDonut commented on VB-Application/src/main/java/com/gmail/visualbukkit/blocks/definitions/ExprPluginAuthors.java:
```diff
@@ -0,0 +1,26 @@
+public class ExprPluginAuthors extends Expression {
+
+    public ExprPluginAuthors() {
+        super("expr-plugin-authors", "Plugin Authors", "Bukkit", "Authors of the plugin");
```
For consistency, "The authors of the plugin"

OfficialDonut commented on VB-Application/src/main/java/com/gmail/visualbukkit/blocks/definitions/StatContinueLoop.java:
```diff
@@ -14,7 +14,7 @@ public Block createBlock() {
             public void update() {
                 super.update();
-                setValid(checkForContainer(StatListLoop.class) || checkForContainer(StatNumberLoop.class) || checkForContainer(StatNumberLoop.class));
+                setValid(checkForContainer(StatListLoop.class) || checkForContainer(StatNumberLoop.class) || checkForContainer(StatAdvancedNumberLoop.class));
```
Please add `checkForContainer(StatWhileLoop.class)`

Sal4iDev replied:
sure

Sal4iDev committed:
Update

Sal4iDev commented:
Done

Sal4iDev committed:
Update GUI.properties

Sal4iDev committed:
Update GUI_ru_RU.properties

Sal4iDev commented:
- Added `(s)` to `label.plugin_author`

Sal4iDev committed:
Update StatAdvancedNumberLoop.java

Sal4iDev commented:
I also added the ability to select the `<`, `<=`, `>`, `>=` mode in the `Advanced Number Loop`

OfficialDonut commented on VB-Application/src/main/resources/lang/GUI.properties:
```diff
@@ -55,7 +55,7 @@ label.enabled_extensions=Enabled Extensions (for current project)
 label.installed_extensions=Installed Extensions
 label.other=Other
-label.plugin_author=Author:
+label.plugin_author=Author(s):
```
https://github.com/OfficialDonut/VisualBukkit/blob/5f891442b066fe850b42c9f18b2be22cfc27ed20/VB-Application/src/main/java/com/gmail/visualbukkit/project/PluginBuilder.java#L232
This has to be changed to support multiple authors

Sal4iDev replied:
Oh, yeah, sure

Sal4iDev replied:
done

Sal4iDev committed:
Update PluginBuilder.java

Sal4iDev committed:
Update PluginBuilder.java
